### DISCUSSION TITLE
How to duplicate an ORM definition?

### DISCUSSION THREAD

**Discussion by z3dev** [sqlalchemy/sqlalchemy#7663] [Usage Questions] [answered]
It's pretty common in the database world to duplicate a table, sometimes temporarily, to select data into for later joins, etc. Is there an easy way to duplicate the ORM definition of an existing table, and create a new ORM for the duplicate table?

Here's my attempt, which works...
```
    # morph the Security ORM into another definition for security updates
    class Update:
        pass

    r = registry()
    t = Table(
        "security_updates",
        r.metadata,
        *([c.copy() for c in Security.__table__.columns]),
    )
    r.map_imperatively(Update, t)
```

**Comment by zzzeek** [comment, marked as answer]
that approach works, although column.copy() is deprecated.   for that exact recipe, use [Table.to_metadata()](https://docs.sqlalchemy.org/en/14/core/metadata.html?highlight=to_metadata#sqlalchemy.schema.Table.to_metadata), which fortunately sometime along the way we added a ".name" argument to:

```py
   r = registry()
   r.map_imperatively(Update, Security.__table__.to_metadata(r.metadata, name="security_updates"))
```

the other way is to define a mixin that has the mapping elements you want, then use that mixin to create copies of the mapping, there's an example from more old school SQLAlchemy versions of that at https://github.com/sqlalchemy/sqlalchemy/wiki/EntityName .
